Add support for Versal Gen 2 APU (Cortex a78 processor)

kedareswararao opened the pull request:
This pull request adds support for Versal Gen 2 APU which based on the ARM A78 processor.

Commit:
soc: amd: Add initial support for Versal Gen 2 SoC APU (Cortex-A78)

Add initial support for the Versal Gen 2 SoC APU, which is based on
the Arm Cortex-A78 processor. It includes basic wiring for memory
regions, UART, interrupt controller, and timer.

The versal2_apu.dtsi file defines peripherals shared across the SoC,
while versal2_a78.dtsi captures peripherals private to the Cortex-A78
processor. These device trees lay the groundwork for further APU-based
development on the Versal Gen 2 platform.

Signed-off-by: Appana Durga Kedareswara rao <appana.durga.kedareswara.rao@amd.com>

Commit:
boards: amd: Add generic APU board support for Versal Gen 2 SoC (CortexA78)

Introduce a generic board configuration for the APU on the Versal Gen 2
SoC, which is based on the ARM Cortex-A78 processor.

This board setup provides a baseline environment for enabling and testing
Cortex-A78 features and peripheral integration on the APU subsystem.

Signed-off-by: Appana Durga Kedareswara rao <appana.durga.kedareswara.rao@amd.com>

michalsimek commented on boards/amd/versal2_apu/versal2_apu-qemu.dts:
That's DT used for QEMU configuration. With Zephyr only devices which are wired with Zephyr DT are tested. Others not but they are modeled. It would be very difficult to prune parts which are not used by Zephyr.
And form of source code is IMHO better than putting DTB or not to provide anything. With this you can run Zephyr on Qemu and have it working directly. Another problem is also that we can't really check that binding against any schema. 

But if you don't like it we don't have an issue not to wire QEMU.

kedareswararao replied:
Thanks @henrikbrixandersen Sure will add a comment

Commit:
arm64: fpu: Clear K_FP_REGS flag in arch_float_disable()

The arch_float_disable() function was not clearing the K_FP_REGS flag
from thread->base.user_options after disabling FPU access. This caused
the float_disable test to fail as it verifies the flag is properly
cleared after FPU disable.

Signed-off-by: Appana Durga Kedareswara rao <appana.durga.kedareswara.rao@amd.com>

Commit:
tests: fpu_sharing: float_disable: Add ARM64 architecture support

The float_disable test was missing proper ARM64 support in its
preprocessor conditionals:

1. ARM64 was not included in K_FP_OPTS definition, causing build errors
2. ARM64 was incorrectly grouped with ARM (Cortex-M/R) which expects
   k_float_disable() to return -EINVAL for non-current threads
3. K_FLOAT_DISABLE_SYSCALL_RETVAL was not defined for ARM64

ARM64 has different FPU management semantics than ARM (Cortex-M/R):
- ARM restricts k_float_disable() to current thread only
- ARM64 allows disabling FPU for any thread because SMP configurations
  require flush_owned_fpu() to manage FPU state across multiple CPUs

Fix the preprocessor conditionals to:
- Include CONFIG_ARM64 in K_FP_OPTS definition
- Separate ARM64 from ARM in test_k_float_disable_common() so ARM64
  expects success (0) instead of -EINVAL
- Include ARM64 in K_FLOAT_DISABLE_SYSCALL_RETVAL definition

Signed-off-by: Appana Durga Kedareswara rao <appana.durga.kedareswara.rao@amd.com>

nordicjm reviewed (approved):
Board/SoC files OK
